# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/quota/ActorsParent.cpp
# Commit: f25dfc157ef7
# Full Hash: f25dfc157ef7d4e336f7625fea7586f19d20bc17
# Author: Jan Varga <jvarga@mozilla.com>
# Date: 2024-10-11 10:49:37
# Regressor Bug: 1866402
# File Overlap Count: 1
# Description:
#   Bug 1866402 - Add QuotaManager::PrincipalMetadataToPrincipalInfo method; r=dom-storage-reviewers,jari
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D195570
# ==============================================================================

diff -r 30841c08def5 -r f25dfc157ef7 dom/quota/ActorsParent.cpp
--- a/dom/quota/ActorsParent.cpp	Thu Oct 10 23:28:44 2024 +0000
+++ b/dom/quota/ActorsParent.cpp	Fri Oct 11 00:06:08 2024 +0000
@@ -6560,6 +6560,27 @@
 }
 
 // static
+Result<PrincipalInfo, nsresult> QuotaManager::PrincipalMetadataToPrincipalInfo(
+    const PrincipalMetadata& aPrincipalMetadata) {
+  QM_TRY_INSPECT(
+      const auto& principal,
+      ([&aPrincipalMetadata]() -> Result<nsCOMPtr<nsIPrincipal>, nsresult> {
+        if (aPrincipalMetadata.mOrigin.EqualsLiteral(kChromeOrigin)) {
+          return nsCOMPtr<nsIPrincipal>(SystemPrincipal::Get());
+        }
+
+        return nsCOMPtr<nsIPrincipal>(
+            BasePrincipal::CreateContentPrincipal(aPrincipalMetadata.mOrigin));
+      }()));
+  QM_TRY(MOZ_TO_RESULT(principal));
+
+  PrincipalInfo principalInfo;
+  QM_TRY(MOZ_TO_RESULT(PrincipalToPrincipalInfo(principal, &principalInfo)));
+
+  return std::move(principalInfo);
+}
+
+// static
 nsAutoCString QuotaManager::GetOriginFromValidatedPrincipalInfo(
     const PrincipalInfo& aPrincipalInfo) {
   MOZ_ASSERT(IsPrincipalInfoValid(aPrincipalInfo));